Beams/Widgets/qMRMLBeamParametersTabWidget.cxx

```diff
@@ -149,12 +153,14 @@ void qMRMLBeamParametersTabWidget::setBeamNode(vtkMRMLNode* node)
   Q_D(qMRMLBeamParametersTabWidget);
 
   vtkMRMLRTBeamNode* beamNode = vtkMRMLRTBeamNode::SafeDownCast(node);
+  vtkMRMLRTIonBeamNode* ionBeamNode = vtkMRMLRTIonBeamNode::SafeDownCast(beamNode);
 
   // Connect display modified event to population of the table
   qvtkReconnect( d->BeamNode, beamNode, vtkCommand::ModifiedEvent,
                  this, SLOT( updateWidgetFromMRML() ) );
 
   d->BeamNode = beamNode;
+  d->IonBeamNode = ionBeamNode;

```

Comment:
I'd strongly prefer storing only one pointer (keep BeamNode and remove IonBeamNode) and doing SafeDownCast at the places of usage to decide what kind of beam it is.

Reply:
Only one pointer left.